controllers/util/secrets_util_test.go

```diff
@@ -490,4 +495,83 @@ var _ = Describe("Secrets_Util", func() {
 			})
 		})
 	})
+	Context("AddSecretToCluster to VeleroNS", func() {
+		When("Secret is initially created in the ramen namespace", func() {
+			Specify("Create the secret", func() {
+				Expect(k8sClient.Create(context.TODO(), secrets[0])).To(Succeed())
+			})
+			It("Returns success", func() {
+				Expect(secretsUtil.AddSecretToCluster(
+					secretNames[0],
+					clusterNames[0],
+					tstNamespace,
+					tstNamespace,
+					util.SecretFormatRamen,
+					"")).To(Succeed())
+			})
+			It("Protects the secret with a finalizer", func() {
+				Expect(finalizerPresent(secretNames[0], util.SecretFormatRamen)).Should(BeTrue())
+			})
+			It("Creates an associated policy for the secret including the cluster", func() {
+				Expect(plRuleContains(plRuleName[0], tstNamespace, clusterNames[:1])).Should(BeTrue())
+				Expect(policyContains(policyName[0], tstNamespace, secrets[0])).Should(BeTrue())
+			})
+		})
+		When("Secret is created in the Velero namespace", func() {
+			It("Succeeds", func() {
+				Expect(secretsUtil.AddSecretToCluster(
+					secretNames[0],
+					clusterNames[0],
+					tstNamespace,
+					tstNamespace,
+					util.SecretFormatVelero,
+					veleroNS)).To(Succeed())
+			})
+			It("Protects the secret with an additional finalizer", func() {
+				Expect(finalizerPresent(secretNames[0], util.SecretFormatVelero)).Should(BeTrue())
+			})
+			It("Creates an associated policy for the secret including the cluster", func() {
+				Expect(plRuleContains(plRuleNameV[0], tstNamespace, clusterNames[:1])).Should(BeTrue())
+				Expect(policyContains(policyNameV[0], tstNamespace, secrets[0])).Should(BeTrue())
+			})
```

Comment:
So we adopt any secret we find in velero namesapce and create a policy for the secret?